File: 415.字符串相加.py
#
# @lc app=leetcode.cn id=415 lang=python
#
# [415] 字符串相加
#

# @lc code=start
class Solution(object):
    def addStrings(self, num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
        res = ""
        i, j, carry = len(num1)-1, len(num2)-1, 0
        while i >= 0 or j >= 0:
            n1 = int(num1[i]) if i >= 0 else 0
            n2 = int(num2[j]) if j >= 0 else 0
            tmp = n1 + n2 + carry
            carry = tmp // 10
            res = str(tmp % 10 ) + res
            i, j = i - 1, j - 1 
        return '1'+ res if carry else res
        
# @lc code=end

File: 415.字符串相加.cpp
/*
 * @lc app=leetcode.cn id=415 lang=cpp
 *
 * [415] 字符串相加
 */

// @lc code=start
class Solution {
public:
    string addStrings(string num1, string num2) {
        int carry = 0;
        string res = "";
        bool flag = false;
        if(num1.size() > num2.size())
        {
            swap(num1, num2);
        }
        while(num1.size() < num2.size()){
            num1.insert(0, "0");
        }
        for(int i = num1.size()-1; i >= 0; i--){
            int tmp = num1[i]-'0' + num2[i] - '0' + carry;
            if(tmp >= 10)
            {
                tmp -= 10;
                carry = 1;
            }
            else{
                carry = 0;
            }
            res += to_string(tmp);
        }
        if(carry)
            res += '1';
        reverse(res.begin(), res.end());
        return res;
    }
};
// @lc code=end

